Traceback (most recent call last):
  File "/opt/hostedtoolcache/Python/3.9.20/x64/lib/python3.9/site-packages/jupyter_cache/executors/utils.py", line 58, in single_nb_execution
    executenb(
  File "/opt/hostedtoolcache/Python/3.9.20/x64/lib/python3.9/site-packages/nbclient/client.py", line 1314, in execute
    return NotebookClient(nb=nb, resources=resources, km=km, **kwargs).execute()
  File "/opt/hostedtoolcache/Python/3.9.20/x64/lib/python3.9/site-packages/jupyter_core/utils/__init__.py", line 165, in wrapped
    return loop.run_until_complete(inner)
  File "/opt/hostedtoolcache/Python/3.9.20/x64/lib/python3.9/asyncio/base_events.py", line 647, in run_until_complete
    return future.result()
  File "/opt/hostedtoolcache/Python/3.9.20/x64/lib/python3.9/site-packages/nbclient/client.py", line 709, in async_execute
    await self.async_execute_cell(
  File "/opt/hostedtoolcache/Python/3.9.20/x64/lib/python3.9/site-packages/nbclient/client.py", line 1062, in async_execute_cell
    await self._check_raise_for_error(cell, cell_index, exec_reply)
  File "/opt/hostedtoolcache/Python/3.9.20/x64/lib/python3.9/site-packages/nbclient/client.py", line 918, in _check_raise_for_error
    raise CellExecutionError.from_cell_and_msg(cell, exec_reply_content)
nbclient.exceptions.CellExecutionError: An error occurred while executing the following cell:
------------------
class ContaBancaria:
    def __init__(self):
        self._saldo = 0

    @property
    def saldo(self):
        return self._saldo

    def depositar(self, valor):
        if valor > 0:
            self._saldo += valor
        else:
            print("Valor inv√°lido para dep√≥sito.")

    def sacar(self, valor):
        if 0 < valor <= self._saldo:
            self._saldo -= valor
        else:
            print("Valor inv√°lido para saque.")


conta = ContaBancaria()
conta.depositar(1000)
# Aqui estamos acessando a propriedade saldo,
# que retorna o valor do atributo privado _saldo
print(conta.saldo)

# Mas por ser uma propriedade, n√£o podemos modificar seu valor diretamente
conta.saldo = 5000
------------------

----- stdout -----
1000
------------------

[0;31m---------------------------------------------------------------------------[0m
[0;31mAttributeError[0m                            Traceback (most recent call last)
Cell [0;32mIn[7], line 29[0m
[1;32m     26[0m [38;5;28mprint[39m(conta[38;5;241m.[39msaldo)
[1;32m     28[0m [38;5;66;03m# Mas por ser uma propriedade, n√£o podemos modificar seu valor diretamente[39;00m
[0;32m---> 29[0m [43mconta[49m[38;5;241;43m.[39;49m[43msaldo[49m [38;5;241m=[39m [38;5;241m5000[39m

[0;31mAttributeError[0m: can't set attribute

